File: frontend/hrms/hrms/src/services/candidateRecourseService.js
import axios from "axios"

export default class CandidateRecourseService{
    add(values){
        return axios.post("http://localhost:8080/api/candidaterecourse/add",values)
    }

    delete(id){
        return axios.delete(`http://localhost:8080/api/candidaterecourse/deletebyid?id=${id}`)
    }


    update(recourse){
        return axios.put(`http://localhost:8080/api/candidaterecourse/update`,recourse)
    }
    getByEmployerId(id){
        return axios.get(`http://localhost:8080/api/candidaterecourse/getbyemployerid?id=${id}`)
    }
    
    getByCandidateId(id){
        return axios.get(`http://localhost:8080/api/candidaterecourse/getbycandidateid?id=${id}`)
    }

    getByCandidateIdAndAdvertId(candidateId,advertId){
        return axios.get(`http://localhost:8080/api/candidaterecourse/getbycandidateandadvert?candidateId=${candidateId}&advertId=${advertId}`)
    }


}

File: backend/Odevler/Gun_7/Odev_1/hrms/src/main/java/javaReact_BootCamp/hrms/business/abstracts/CandidateRecourseService.java
package javaReact_BootCamp.hrms.business.abstracts;

import java.util.List;

import javaReact_BootCamp.hrms.core.utilites.result.DataResult;
import javaReact_BootCamp.hrms.core.utilites.result.Result;
import javaReact_BootCamp.hrms.entities.concretes.CandidateRecourse;
import javaReact_BootCamp.hrms.entities.dtos.RecourseWithEmployerAndCandidate;

public interface CandidateRecourseService {
	
	DataResult<List<CandidateRecourse>> getAll();
	
	DataResult<List<CandidateRecourse>> getByCandidateId(int id);
	
	DataResult<List<CandidateRecourse>> getByEmployerId(Integer id);
	
	DataResult<List<CandidateRecourse>> getByAdvertId(Integer id);
	
	DataResult<List<CandidateRecourse>> getByStatus(String status);
	
	DataResult<CandidateRecourse> getByCandidateIdAndAdvertId(Integer candidateId,Integer advertId );
	
	Result add(CandidateRecourse recourse);
	
	Result delete(int id);
	
	Result update(CandidateRecourse recourse);
	

}
